### PR TITLE
Feature/modern components

### PR DESCRIPTION
**Pull request by borysyuk**

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -156,13 +156,13 @@ export class Api {
     });
   }
 
-  aggregate(endpoint, cube, params) {
+  aggregate_old(endpoint, cube, params) {
```
Comment (pwalsh) [inline src/api/index.js]: Why `aggregate_old`? Is this test code?

File: src/components/geoview/api.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+require('isomorphic-fetch');
+
+var cosmopolitanApiUrl = 'http://cosmopolitan.openspending.org/';
+
+function loadGeoJson(countryCode) {
+  // Uncomment when cosmopolitan API implemented
+  //var url = cosmopolitanApiUrl + '/geojson/country/' +
+  //  encodeURIComponent(countryCode) + '?format=json';
+
+  var url = 'https://rawgit.com/openspending/fiscal-data-package-demos/' +
```
Comment (pwalsh) [inline src/components/geoview/api.js]: This can't be here, and should be changed based on discussed hack.

**Comment by pwalsh** [comment]
@borysyuk this all looks good, just address the hack for polygons, and the question about `aggregate_old`

File: src/bindings/angular/geoview/directive.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            resizeHandlers.forEach(function(callback) {
+              $window.removeEventListener('resize', callback);
+            });
+            resizeHandlers = [];
+          }
+
+          function createHandle(countryCode, data, currencySign) {
+            api.loadGeoJson($scope.cosmoEndpoint, countryCode)
+              .then(function(geoJson) {
+                console.log('===============');
```
Comment (pwalsh) [inline src/bindings/angular/geoview/directive.js]: Please remove these (and other) called to `console.log`

File: src/components/bubbletree/index.js [hunk 4]
```diff
@@ -11,17 +13,40 @@ export class BubbleTreeComponent extends events.EventEmitter {
   constructor() {
     super();
     this.wrapper = null;
-    this.chart = null;
+    this.bubbleTreet = null;
```
Comment (pwalsh) [inline src/components/bubbletree/index.js]: Typo

File: src/components/bubbletree/index.js [hunk 5]
```diff
@@ -11,17 +13,40 @@ export class BubbleTreeComponent extends events.EventEmitter {
...
   }
 
   refresh() {
-    var bounds = this.wrapper.getBoundingClientRect();
-    this.chart.resize({
-      height: bounds.height,
-      width: bounds.width
-    });
+    //var bounds = this.wrapper.getBoundingClientRect();
```
Comment (pwalsh) [inline src/components/bubbletree/index.js]: What is this commented code for?

File: src/components/chart/index.js [hunk 6]
```diff
@@ -23,27 +23,82 @@ export class ChartComponent extends events.EventEmitter {
...
+              if (chartType == 'bar') {
+                c = d.index;
+              };
+              return Utils.colorScale(c);
+            },
+            type: chartType || 'bar',
+            x: _.first(_.first(columns)),
+            groups: [[columns[1][0]]],
+            types: types,
+            //onclick: (d, element) => {
```
Comment (pwalsh) [inline src/components/chart/index.js]: Do we need this commented code?

File: src/components/index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import BubbleTreeComponent from './bubbletree/index'
```
Comment (pwalsh) [inline src/components/index.js]: You do not need `/index` on imports

File: src/components/sankey/index.js [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+  constructor() {
+    super();
+    this.wrapper = null;
+    this.sankey = null;
+  }
+
+  refresh() {
+    var bounds = this.wrapper.getBoundingClientRect();
+    //this.chart.resize({
```
Comment (pwalsh) [inline src/components/sankey/index.js]: ?

File: src/components/treemap/index.js [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import Api from '../../api'
+import { Api } from '../../api/index'
```
Comment (pwalsh) [inline src/components/treemap/index.js]: Dont need `/index`

File: src/index.js [hunk 10]
```diff
@@ -1,14 +1,20 @@
...
-require('./chart')
-require('./crosstab')
-require('./facts')
-require('./pager')
-require('./panel')
-require('./sankey')
-require('./treemap')
-require('./util')
-require('./workspace')
+//require('./app')
```
Comment (pwalsh) [inline src/index.js]: Why all this commented code?

**Comment by pwalsh** [comment]
@borysyuk some more comments now, please address then i will merge.

File: src/index.js (hunk 10 above)
Comment (borysyuk) [inline src/index.js]: there are old versions of babbage.ui components

Comment (pwalsh) [inline src/index.js]: I know. but why do we need them left as commented out code?

Comment (borysyuk) [inline src/index.js]: I can delete all old components if we will use our fork only. New components are a library but old code base is an application. So new code is not equal to old one.

Comment (pwalsh) [inline src/index.js]: @borysyuk please remove *all* code we are no longer using. Do a new PR for that.

**Pull request merged**